flowexperimental/comp/wells/CompWellInterface.hpp

```diff
@@ -98,12 +108,12 @@ class CompWellInterface
     std::vector<int> saturation_table_number_;
 
     // std::string name_;
-
 };
 
-
 } // end of namespace Opm
 
+#ifndef OPM_COMP_WELLINTERFACE_IMPL_HPP
```

Comment:
This introduced extra lines, do we need it since the `#ifndef OPM_COMP_WELLINTERFACE_IMPL_HPP` defined in `CompWellInterface_impl.hpp`

Reply:
yes. this is for IDEs only. it needs to include the CompWellInterface.hpp to have the symbols defined. in regular compilation the _impl.hpp file is included from the .hpp file. For IDEs it's the opposite, the _impl.hpp includes the .hpp file. These lines are needed to avoid circular includes in that case.

Reply:
What IDE are you using and do you need to set some specifications to see this type of warning?

Reply:
I'm using qtcreator, but anything using clangd or intellisense or whatever will show the issue. It creates the AST of the _impl.hpp file to analyze, and when it does, it has no definition of the class available.

Reply:
> This introduced extra lines, do we need it since the `#ifndef OPM_COMP_WELLINTERFACE_IMPL_HPP` defined in `CompWellInterface_impl.hpp`

For what it's worth, this is the structure we (ultimately) ended up using following the discussion in [#5834](https://github.com/OPM/opm-simulators/pull/5834#discussion_r1900899194) and also implemented in #5835.